Installation issue: ROOT makes me sad

### Steps to reproduce the issue

On my latest macOS reinstall, ROOT fails to build because its vendored `afterimage` dependency includes a vendored `libungif` dependency which apparently requires x11, even if x11 is disabled at the ROOT level. Adding an always-on `libx11` dependency doesn't seem to fix it. :(

```console
$ spack install root ~aqua ~davix ~examples ~opengl ~x ~tbb ~rootfit ~math ~gsl cxxstd=14
```

From the ROOT cmake step:
```
-- Enabled support for:  builtin_afterimage builtin_clang builtin_cling builtin_llvm dataframe exceptions gdml libcxx minuit2 pyroot roofit rpath shared soversion unuran vdt xml
```

Build error:
```
CMake Error at /private/var/folders/gy/mrg1ffts2h945qj9k29s1l1dvvmbqb/T/s3j/spack-stage/spack-stage-root-6.24.00-4kle6fzd3mi546sbzli5vwbsi3u7rmhc/spack-build-4kle6fz/AFTERIMAGE-prefix/src/AFTERIMAGE-stamp/AFTERIMAGE-build-RelWithDebInfo.cmake:49 (message):
  Command failed: 2

   '/Applications/Xcode.app/Contents/Developer/usr/bin/make'

  See also

    /private/var/folders/gy/mrg1ffts2h945qj9k29s1l1dvvmbqb/T/s3j/spack-stage/spack-stage-root-6.24.00-4kle6fzd3mi546sbzli5vwbsi3u7rmhc/spack-build-4kle6fz/AFTERIMAGE-prefix/src/AFTERIMAGE-stamp/AFTERIMAGE-build-*.log


make[2]: *** [AFTERIMAGE-prefix/src/AFTERIMAGE-stamp/AFTERIMAGE-build] Error 1
make[1]: *** [CMakeFiles/AFTERIMAGE.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```
and in the log file that it points to:
```
In file included from libungif/dgif_lib.c:37:
In file included from libungif/gif_lib_private.h:5:
In file included from libungif/gif_hash.h:24:
In file included from libungif/../afterbase.h:36:
In file included from libungif/../asim_afterbase.h:23:
libungif/../xwrap.h:5:11: fatal error: 'X11/Xlib.h' file not found
# include <X11/Xlib.h>
          ^~~~~~~~~~~~
1 error generated.
make[3]: *** [libungif/dgif_lib.o] Error 1
```

### Information on your system

$ spack debug report
* **Spack:** 0.16.2-2855-35f7959f54
* **Python:** 3.8.5
* **Platform:** darwin-bigsur-skylake
* **Concretizer:** clingo
* 
### Additional information

- [spack-build-env.txt](https://github.com/spack/spack/files/6555335/spack-build-env.txt)
- [spack-build-out.txt](https://github.com/spack/spack/files/6555339/spack-build-out.txt)
- [spack-build-01-cmake-out.txt](https://github.com/spack/spack/files/6555340/spack-build-01-cmake-out.txt)
- [spack-build-02-build-out.txt](https://github.com/spack/spack/files/6555341/spack-build-02-build-out.txt)
- [AFTERIMAGE-build-err.log](https://github.com/spack/spack/files/6555342/AFTERIMAGE-build-err.log)
- [AFTERIMAGE-build-out.log](https://github.com/spack/spack/files/6555343/AFTERIMAGE-build-out.log)

@HadrienG2 @chissg @drbenmorgan  @vvolkl

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/root/package.py
+++ b/var/spack/repos/builtin/packages/root/package.py
@@ -359,7 +359,7 @@ class Root(CMakePackage):
         # Options related to ROOT's ability to download and build its own
         # dependencies. Per Spack convention, this should generally be avoided.
         options += [
-            define('builtin_afterimage', True),
+            define_from_variant('builtin_afterimage', 'x'),
             define('builtin_cfitsio', False),
             define('builtin_davix', False),
             define('builtin_fftw3', False),

